src/test/groovy/co/elastic/TestUtils.groovy

```diff
@@ -20,14 +20,18 @@ package co.elastic
 public class TestUtils {
 
   public static withEnvInterceptor = { list, closure ->
-    list.forEach {
-      def fields = it.split("=")
-      binding.setVariable(fields[0], fields[1])
+    def savedVars = []
+    // filter invlaid entries - with name empty or containing only spaces
+    list.findAll{ it&&it=~/^\s*[^=\s]+\s*=?/ }.forEach{
+      List fields = it.split("=");
+      def name = fields.remove(0);
+      def value = fields.join("="); // add back = chars deleted by split
+      savedVars.add([name, binding.hasVariable(name) ? binding.getVariable(name) : null])
+      binding.setVariable(name, value.size()> 0 ? value : null)
     }
     def res = closure.call()
-    list.forEach {
-      def fields = it.split("=")
-      binding.setVariable(fields[0], null)
+    savedVars.reverse().forEach {
+      binding.setVariable(it[0], it[1])
```

Comment:
What's the reason for this particular change?

Reply:
The reason for this is to correctly restore variable value if this variable specified few times in arguments to withEnv.
LIke this:
```
withEnv(["VAR1=AAA","VAR2=BBB","VAR1=CCC"]) 
```
May be not a big improvement bit it was easy to do and Jenkins also works this way.

Reply:
Gotcha! Thanks for the clarification